yt_dlp/downloader/external.py

```diff
@@ -302,6 +316,137 @@ def _make_cmd(self, tmpfilename, info_dict):
             cmd += ['--', info_dict['url']]
         return cmd
 
+    def _call_downloader(self, tmpfilename, info_dict):
+        info_dict.pop('__rpc_port', None)
+        info_dict.pop('__rpc_secret', None)
+        if self._ENABLE_PROGRESS:
+            import uuid
+            info_dict = info_dict.copy()
+            info_dict['__rpc_port'] = find_available_port() or 19190
+            info_dict['__rpc_secret'] = str(uuid.uuid4())
```

Comment:
```suggestion
        info_dict['__rpc_port'] = find_available_port() or 19190
        info_dict['__rpc_secret'] = str(uuid.uuid4())
```
* why's _ENABLE_PROGRESS needed?
* why the pop and copy?
* Why import uuid locally?

Reply:
> why's _ENABLE_PROGRESS needed?
> why the pop and copy?

I don't remember, ytdl-patched doesn't have it right now (and `__rpc_secret` is gone)

> Why import uuid locally?

maybe top reduce memory usage or to reduce the chance to conflict